File: CSharp - Monogame SDK 3.7/Explosion.cs
using System.Collections.Generic;

namespace Asteroids {

    public class Explosion : Entity {

        private Explosion() { }

        public static void Spawn(Animation anim, double x, double y) {
            Queue<Explosion> pool = new Queue<Explosion>();

            Explosion explosion = (pool.Count == 0 || pool.Peek().active) ? new Explosion() : pool.Dequeue();
            explosion.anim = anim;
            explosion.frame = 0;
            explosion.x = x;
            explosion.y = y;
            explosion.looped = false;
            explosion.active = true;

            pool.Enqueue(explosion);
            Astro.Instance.entities.Add(explosion);
        }

        override public void Update() {
            base.Update();
            if (looped) active = false; // remove when the animation completes
        }
    }
}

File: Python - pygame/Explosion.py
from Entity import *
from collections import deque

class Explosion(Entity):
  pool = deque()

  @staticmethod
  def spawn(anim, x, y):
    asset = Explosion() if len(Explosion.pool) == 0 or Explosion.pool[0].active else Explosion.pool.popleft()
    asset.anim = anim
    asset.x = x
    asset.y = y
    asset.looped = False
    asset.active = True

    Explosion.pool.append(asset)
    Global.entities.append(asset)

  def update(self):
    super().update()
    if self.looped: self.active = False # remove when the animation completes
